deliciousbrains/wp-post-promoter
# chore(WEB-6711): Use wp-scripts instead of gulp for build

This PR replaces Gulp with [wp-scripts](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/) and refactors the assets to work with the default wp-scripts config.

Now:

* `yarn build` or `npx wp-scripts build` builds the assets for "production"
* `yarn start` or `npx wp-scripts start` build the assets for development (with sourcemaps) and watches for changes

I renamed `social.scss` to `social-styles.scss` because otherwise there was a name clash with `social.js` in the output.

## Testing

This package makes the "Email Draft" and "Social Promoter" meta boxes on post edit pages work.

<img width="1432" alt="Screenshot 2023-04-13 at 10 33 19" src="https://user-images.githubusercontent.com/1532368/231718217-4abd5ab8-690e-4f5d-b798-920aa905a5f0.png">

The styles and scripts are pretty simple and don't do very much. So to test:

1. Pull the change to a local copy of the `wp-post-promoter` package
2. Update the composer.json in a local copy of a site that uses the package to include the following in the `repositories` key:

```
    {
      "type": "path",
      "url": "/your/path/to/wp-post-promoter"
    },
```

3. Run `composer update deliciousbrains/wp-post-promoter`. It should say:

```
  - Removing deliciousbrains/wp-post-promoter (dev-master be16473)
  - Installing deliciousbrains/wp-post-promoter (dev-master): Symlinking from /your/path/to/wp-post-promoter
```

4. Log in to your local copy of the site that uses this package and edit a post in the classic editor.
5. Check the JS console in dev tools for errors related to the change
6. Check that the two meta boxes "Email Draft" and "Social Promoter" are styled correctly - compare with the live site it necessary, but failed CSS should be pretty obvious.
7. Change the value in the "Campaign" input in the "Social Promoter" meta box and click "Update" - this should update the `utm_campaign` parameter in the links underneath.
8. Click the Twitter icon and make sure that a pop-up Tweet box appears.
9. Check the console for errors again, but if everything worked then the assets loaded OK.
